examples/generator/bernstein_vazirani.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+# Copyright(c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+from typing import List, Callable, Tuple
+from pyqir_generator import QirBuilder
+
+
+ParityFunction: type = Callable[[Tuple[List[str], str]], None]
+
+
+class BernsteinVazirani:
+    """
+    Implementation of the Bernstein-Vazirani quantum algorithm
+    """
+
+    def __init__(self, qubit_count: int = 8, pattern: int = 12):
+        """
+        :param qubit_count: number of qubits to use
+        :type qubit_count: int
+        :param pattern: integer representation of the bitstring pattern
+        :type pattern: int
+        """
+        self.builder = QirBuilder("Bernstein-Vazirani")
```

Comment:
Given that you annotate types, is it worthwhile to annotate the class members as well? I.e. 
```python
class BernsteinVazirani:
    """
    Implementation of the Bernstein-Vazirani quantum algorithm
    """
    builder: QirBuilder
    
    # ...
```